src/filehandler/pemhandler.cpp

```diff
@@ -1,40 +1,50 @@
 #include "filehandler/pemhandler.h"
+#include "params.h"
 #include "base64.h"
 #include "hex.h"
 
 #include <fstream>
 #include <iostream>
 #include <sodium.h>
+#include <stdexcept>
+#include <iomanip>
 
 namespace ih
 {
-PemFileReader::PemFileReader(std::string const filePath) :
+PemFileReader::PemFileReader(std::string const &filePath) :
         IFileHandler(filePath)
 {
-    m_fileContent = getFileContent();
+    if (PemFileReader::isFileValid())
+    {
+        std::string const fileContent = getFileContent();
+        if (fileContent.empty())
+            throw std::invalid_argument(ERROR_MSG_FILE_EMPTY);
+
+        m_fileKeyBytes = getKeyBytesFromContent(fileContent);
+        if(m_fileKeyBytes.size() != (crypto_sign_PUBLICKEYBYTES + crypto_sign_SEEDBYTES))
+            throw std::length_error(ERROR_MSG_KEY_BYTES_SIZE);
+    }
 }
 
 bool PemFileReader::isFileValid() const
 {
-    return IFileHandler::fileExists() &&
-            IFileHandler::isFileExtension("pem") &&
-           (m_fileContent != "");
+    bool const fileExists = IFileHandler::fileExists();
+    bool const fileExtensionValid = IFileHandler::isFileExtension("pem");
+
+    if (!fileExists) throw std::invalid_argument(ERROR_MSG_FILE_DOES_NOT_EXIST);
+    if (!fileExtensionValid) throw std::invalid_argument(ERROR_MSG_FILE_EXTENSION_INVALID);
+
+    return (fileExists && fileExtensionValid) ;
```

Comment:
... `return true`? Remove return value, actually, since all non-happy cases result in exception? And rename to `checkFile()`? `checkFileExists` + `checkFileExtension`.

Reply:
Yes, that's right, returning (fileExists && fileExtensionValid) makes no sense, I've missed it :(
I've changed the method name from _isFileValid -> checkFile_.
However, IMHO `fileExists` and `isFileExtension` are better names.If you disagree, we can change it